### ISSUE TITLE
[MIG-518]: Add ability to configure custom route selector on rsyncd route

### ISSUE THREAD

**Issue by alaypatel07** [konveyor/mig-controller#919]
**Is your feature request related to a problem? Please describe.**
For users running more than one OpenShift routers, we need a way to select which router will be used for the rsyncd routes on the destination cluster. The way we do that is ask the customer to configure 1 router using [routeSelector](https://docs.openshift.com/container-platform/4.6/networking/ingress-operator.html#nw-ingress-sharding-route-labels_configuring-ingress). After that the customer will edit a value on migrationcontroller CR and start the migration. 

The mig-controller pod will get the config map value, and will slap a label on each rsyncd route it creates.

**Describe the solution you'd like**
1. expose a configuration knob: `rsyncd_route_selector` on migrationcontroller CR
1. Since the route selector can be multiple key value, the mig-operator will convert it into a JSON string and slap it on configmap migration-controller. 
1. mig-operator will restart mig-controller pod
1. mig-controller will see the change and start adding those values to the rsyncd route.

**Describe alternatives you've considered**
-

**Additional context**
This is an urgent requirement for a major potential user

**Comment by shawn-hurley** [comment]
I would like to make sure that:
1. rsyncd_route_selector can handle all the types of selectors, I think it should be a yaml blob as the input
2. An enhancement should be created to add a specific CR field to MigPlan. 

Other than that this LGTM

**Comment by alaypatel07** [comment]
> rsyncd_route_selector can handle all the types of selectors, I think it should be a yaml blob as the input

yes, thats the plan, take yaml, convert it into JSON to keep the semantics intact in the config map, and then use it in the controller.
